CONTRIBUTING.md

```diff
@@ -0,0 +1,129 @@
+When applied to an example, you can access any value in the example dictionary
+via its key. If you just want to print that value surround it in double curly
+braces. For example, if you want to print a value with the key `text`, use this:
+```jinja2
+The text in this example is {{ text }}.
+```
+
+You can also use information from the example to control behavior. For example,
+suppose we have a label with the key `label` in our example, which either has a
+value of 0 or 1. That's not very "natural" language, so maybe we want to decide
+which label name to use based on the example. We can do this by creating a list
+and indexing it with the example key:
+```jinja2
+The label for this example is {{ ["Label A", "Label B"][label] }}.
+```
+We can also use dictionaries for the same thing:
+```jinja2
+The label for this example is {{
+{"a": "Label A",
+ "b": "Label B"
+}[label]
+}}.
+```
+
+Note that some things in a template are particular to the task, and should not be
+modified by downstream steps that try to increase the diversity of the prompts.
+A common example is listing label names in the prompt to provide choices. Anything
+that should not be modified by data augmentation should be surrounded by double
+curly braces and quoted. For example:
+```jinja2
+The choices are {{"a"}}, {{"b"}}, and {{"c"}}.
+```
+
+Finally, remember that a template must produce two strings: a prompt and an output.
+To separate these two pieces, use three vertical bars `|||`.
+So, a complete template for AG News could be:
+```jinja2
+{{text}} 
+Is this a piece of news regarding {{"world politics"}}, {{"sports"}}, {{"business"}}, {{"science and technology"}}? ||| 
+{{ ["World politics", "Sports", "Business", "Science and technology"][label] }}
+```
+   
+## Best Practices
+
+A few miscellaneous things:
+
+* **Writing outputs.** When writing a template for an task that requires outputting
+a label, don't use articles or other stop words before the label name in the output.
+For example, in TREC, the output should be "Person", not "A person". The reason
+is that evaluations often look at the first word of the generated output to determine
+correctness.
+*  **Skipping datasets.** You might find a dataset in the spreadsheet that it
+doesn't make sense to write templates for. For example, a dataset might just be
+text without any annotations. For other cases, ask on Slack. If skipping a dataset,
+mark it in red on the spreadsheet.
+
+## Uploading Templates
+
+Once you save or modify a template, the corresponding file inside the `templates`
+directory in the repo will be modified. To upload it, following these steps:
+1. Run `make style` and `make quality`.
+2. Commit the modified template files (anything under `templates`) to git.
+3. Push to your fork on GitHub.
+4. Open a pull request against `main` on the PromptSource repo.
+5. When the PR is merged into main, mark the dataset in green on the spreadsheet. 
+
+
+## Jinja Cookbook
+TODO
```

Comment:
Maybe say something like 

"Jinja includes lots of complex features but for most instances you should only need to use ... "

We can have people ping the channel if they want to do crazy things.